stl/src/atomic_wait.cpp

```diff
@@ -0,0 +1,249 @@
+        return result;
+    }
+
+
+#if _STL_WIN32_WINNT >= _WIN32_WINNT_WIN8
+    constexpr bool _Have_wait_functions() {
+        return true;
+    }
+#define __crtWaitOnAddress       WaitOnAddress
+#define __crtWakeByAddressSingle WakeByAddressSingle
+#define __crtWakeByAddressAll    WakeByAddressAll
+
+#pragma comment(lib, "Synchronization.lib")
+
+    [[noreturn]] void _Atomic_wait_fallback(
+        [[maybe_unused]] const void* const _Storage, [[maybe_unused]] unsigned long long& _Wait_context) noexcept {
+        std::terminate();
+    }
+
+    [[noreturn]] void _Atomic_notify_fallback([[maybe_unused]] const void* const _Storage) noexcept {
+        std::terminate();
+    }
+
+    void _Atomic_unwait_fallback(
+        [[maybe_unused]] const void* const _Storage, [[maybe_unused]] unsigned long long& _Wait_context) noexcept {}
+
+#else // ^^^ _STL_WIN32_WINNT >= _WIN32_WINNT_WIN8 / _STL_WIN32_WINNT < _WIN32_WINNT_WIN8 vvv
+    void _Atomic_wait_fallback(const void* const _Storage, unsigned long long& _Wait_context) noexcept {
+        switch (_Wait_context & _Atomic_wait_phase_mask) {
+        case _Atomic_wait_phase_wait_none: {
+            _Wait_context = _Atomic_wait_phase_wait_locked;
+            auto& _Entry  = _Atomic_wait_table_entry(_Storage);
+            ::AcquireSRWLockExclusive(&_Entry._Lock);
+            // re-check, and still in _Atomic_wait_phase_wait_locked
+            break;
+        }
+
+        case _Atomic_wait_phase_wait_locked: {
+            auto& _Entry = _Atomic_wait_table_entry(_Storage);
+            ::SleepConditionVariableSRW(&_Entry._Condition, &_Entry._Lock, INFINITE, 0);
+            // re-check, and still in _Atomic_wait_phase_wait_locked
+            break;
+        }
+        }
+    }
+
+    void _Atomic_unwait_fallback(const void* const _Storage, unsigned long long& _Wait_context) noexcept {
+        if (_Wait_context & _Atomic_wait_phase_wait_locked) {
+            auto& _Entry = _Atomic_wait_table_entry(_Storage);
+            ::ReleaseSRWLockExclusive(&_Entry._Lock);
+        }
+    }
+
+    void _Atomic_notify_fallback(const void* const _Storage) noexcept {
+        auto& _Entry = _Atomic_wait_table_entry(_Storage);
+        ::AcquireSRWLockExclusive(&_Entry._Lock);
+        ::ReleaseSRWLockExclusive(&_Entry._Lock);
+        ::WakeAllConditionVariable(&_Entry._Condition);
+    }
+
+    struct _Wait_on_address_functions {
+        std::atomic<decltype(&::WaitOnAddress)> _Pfn_WaitOnAddress;
+        std::atomic<decltype(&::WakeByAddressSingle)> _Pfn_WakeByAddressSingle;
+        std::atomic<decltype(&::WakeByAddressAll)> _Pfn_WakeByAddressAll;
+        std::atomic<bool> _Initialized;
+    };
+
+    const _Wait_on_address_functions& _Get_wait_functions() {
+        static _Wait_on_address_functions functions;
+        if (!functions._Initialized.load(std::memory_order_acquire)) {
+            HMODULE sync_api_module        = ::GetModuleHandle(TEXT("API-MS-WIN-CORE-SYNCH-L1-2-0.DLL"));

```

Comment:
Is `TEXT` a macro for `L"meow"` wide string literals?

Reply:
Yes. I can use L.
Additionally I can explicitly call `GetModuleHandleW`
What are conventions for that?

Reply:
`parallel_algorithms.cpp` uses L and W, I'm now following that.